packages/@lightningjs/ui-components/src/mixins/withTextMagnifier/index.js

```diff
@@ -0,0 +1,78 @@
+import lng from '@lightningjs/core';
+import TextMagnifier from '../../components/TextMagnifier/TextMagnifier';
+
+function isSubclass(Subclass, Superclass) {
+  // Traverse up the prototype chain to check if the Superclass is found
+  while (Subclass) {
+    if (Subclass === Superclass) {
+      return true;
+    }
+    Subclass = Object.getPrototypeOf(Subclass);
+  }
+  return false;
+}
+
+export default function (Base) {
+  if (!isSubclass(Base, lng.Application)) {
+    console.error(
+      'withTextMagnifier can only be applied to an application class'
+    );
+    return Base;
+  }
+
+  return class extends Base {
+    set textMagnifierEnabled(val) {
+      this._textMagnifierEnabled = val;
+      this._updateTextMagnifier(true);
+    }
+
+    get textMagnifierEnabled() {
+      return this._textMagnifierEnabled;
+    }
+
+    get _focusedElement() {
+      return this.application?.focusPath?.[
+        this.application.focusPath.length - 1
+      ];
+    }
+
+    get _location() {
+      const { py: focusedY } = this._focusedElement?.core?.renderContext || {};
+      return focusedY > this.stage.h / 2 ? 'top' : 'bottom';
+    }
+
+    get _focusText() {
+      const { title, description, announce } = this._focusedElement || {};
```

Comment:
I know this is endlessly expandable, so do we maybe want to make it a property users can define strings for and we can search for those on the focused element? Maybe this is the default? I'm just thinking that TextBox and InlineContent both have `content`, which I know `announce` would capture, but what if the user overrode that for whatever reason? I think offering the flexibility could come in handy.

Reply:
You can now set a focusText method on the application and it will use that. What do you think?